tests/map-functions/sameterm-03.rq

```diff
@@ -2,5 +2,6 @@ PREFIX cdt:    <http://w3id.org/awslabs/neptune/SPARQL-CDTs/>
 PREFIX xsd:    <http://www.w3.org/2001/XMLSchema#>
 
 ASK {
-	FILTER(! SAMETERM("{1: 2}"^^cdt:Map, "{1: '2'^^<http://www.w3.org/2001/XMLSchema#integer>}"^^cdt:Map))
+	BIND( cdt:Map(1, '2'^^<http://www.w3.org/2001/XMLSchema#integer>) AS ?map2 )
+	FILTER(! SAMETERM("{1: 2}"^^cdt:Map, ?map2) )
```

Comment:
This one may be a problem for systems that do support CDT literals but don't keep around the lexical forms of these literals (but, instead, turn them into some internal representation of the corresponding value form). When such systems bind `?map2` to the cdt:Map literal that represents the term map with the key-value pair (1,2) and then create a lexical form for that literal---for the purpose of the SAMETERM comparison---they may create a lexical form that happens to be the same as the lexical form of the other cdt:Map literal in the FILTER.

To avoid this problem I propose to switch the two literals as follows.
```suggestion
	BIND( cdt:Map(1, 2) AS ?map1 )
	FILTER(! SAMETERM(?map1, "{1: '2'^^<http://www.w3.org/2001/XMLSchema#integer>}"^^cdt:Map) )
```

Reply:
Good catch. I did see this same scenario in other tests but miss this one. Thanks.